libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -342,7 +342,17 @@ impl<'a, 'b> LiftAstToDml<'a, 'b> {
                     native_type_attribute.span,
                 ))
             } else {
-                Ok((dml::FieldType::Base(scalar_type, type_alias), vec![]))
+                match scalar_type {
+                    ScalarType::Decimal if !supports_native_types => Err(DatamodelError::new_connector_error(
+                        &ConnectorError::from_kind(ErrorKind::NativeFlagsPreviewFeatureDisabled).to_string(),
+                        ast_field.span,
+                    )),
+                    ScalarType::Bytes if !supports_native_types => Err(DatamodelError::new_connector_error(
+                        &ConnectorError::from_kind(ErrorKind::NativeFlagsPreviewFeatureDisabled).to_string(),
+                        ast_field.span,
+                    )),
+                    _ => Ok((dml::FieldType::Base(scalar_type, type_alias), vec![])),
+                }
```

Comment:
I tried to look at the context and the very long function is scary to me,but this specific diff looks good :+1:

Reply:
yes that function is on my list to clean up already 😅